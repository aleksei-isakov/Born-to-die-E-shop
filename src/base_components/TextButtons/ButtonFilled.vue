<template>
  <BaseTextButton
    class="button-filled"
    :class="color"
    @click="onClickEmitEvent"
  >
    <slot></slot>
  </BaseTextButton>
</template>

<script>
import { BaseTextButton } from '..';

export default {
  name: 'ButtonFilled',

  components: {
    BaseTextButton
  },

  props: {
    color: {
      type: String,
      default: 'blue'
    }
  },

  methods: {
    onClickEmitEvent() {
      this.$emit('click');
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';

.button-filled {
  color: $white;
  box-shadow: $shadow;
  &.blue {
    background-color: $primary;
    &:hover {
      background-color: $primary-hover;
    }
  }
  &.red {
    background-color: $error;
    &:hover {
      background-color: $error-hover;
    }
  }
  &:disabled {
    background-color: $button-disabled-bg;
    color: $button-disabled-color;
    pointer-events: none;
  }
}
</style>
