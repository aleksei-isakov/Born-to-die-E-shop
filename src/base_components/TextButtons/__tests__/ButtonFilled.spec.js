import { shallowMount, createLocalVue } from '@vue/test-utils';
import ButtonFilled from '../ButtonFilled.vue';

describe('ButtonFilled.vue', () => {
  let wrapper;

  beforeEach(() => {
    const localVue = createLocalVue();

    wrapper = shallowMount(ButtonFilled, {
      localVue
    });

    wrapper.vm.onClickEmitEvent();
    expect(wrapper.find('button')).toBeTruthy();
    expect(wrapper.is(ButtonFilled)).toBeTruthy();
  });

  it('renders a valid snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
