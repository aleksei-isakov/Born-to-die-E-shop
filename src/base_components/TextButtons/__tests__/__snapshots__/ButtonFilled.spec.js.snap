// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFilled.vue renders a valid snapshot 1`] = `
<basetextbutton-stub
  class="button-filled blue"
/>
`;
